File: MovieTutorial/MovieTutorial.Web/Modules/MovieDB/MovieGenres/MovieGenresColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MovieTutorial.MovieDB.Columns;

[ColumnsScript("MovieDB.MovieGenres")]
[BasedOnRow(typeof(MovieGenresRow), CheckNames = true)]
public class MovieGenresColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MovieGenreId { get; set; }
    public string MovieTitle { get; set; }
    public string GenreName { get; set; }
}

File: MovieTutorial/MovieTutorial.Web/Modules/ServerTypes/MovieDB/MovieGenresColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MovieGenresRow } from "./MovieGenresRow";

export interface MovieGenresColumns {
    MovieGenreId: Column<MovieGenresRow>;
    MovieTitle: Column<MovieGenresRow>;
    GenreName: Column<MovieGenresRow>;
}

export class MovieGenresColumns extends ColumnsBase<MovieGenresRow> {
    static readonly columnsKey = 'MovieDB.MovieGenres';
    static readonly Fields = fieldsProxy<MovieGenresColumns>();
}